libtheora: Fix compilation when MACOSX_DEPLOYMENT_TARGET=11

jdelStrother opened the pull request:
MACOSX_DEPLOYMENT_TARGET will be 11.0 when built in arm64. libtheora's
libtool version doesn't handle this well, resulting in the linking phase
missing the "-Wl,-undefined -Wl,dynamic_lookup" args : 

```
libtheora> Undefined symbols for architecture arm64:
libtheora> "_th_comment_add", referenced from:
libtheora> _theora_comment_add in apiwrapper.o
libtheora> "_th_comment_add_tag", referenced from:
libtheora> _theora_comment_add_tag in apiwrapper.o
libtheora> "_th_comment_clear", referenced from:
libtheora> _theora_comment_clear in apiwrapper.o
libtheora> "_th_comment_init", referenced from:
libtheora> _theora_comment_init in apiwrapper.o
libtheora> "_th_comment_query", referenced from:
libtheora> _theora_comment_query in apiwrapper.o
libtheora> "_th_comment_query_count", referenced from:
libtheora> _theora_comment_query_count in apiwrapper.o
libtheora> ld: symbol(s) not found for architecture arm64
libtheora> clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
```

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

I've been building packages on an M1 mac using @thefloweringash's apple-silicon branch (https://github.com/NixOS/nixpkgs/pull/105026). With this change I can get libtheora & ffmpeg building on arm.

~By itself this fix doesn't really help anything, but rather than add to the amount of stuff to be reviewed in https://github.com/NixOS/nixpkgs/pull/105026 I thought it might be better to submit this separately?~

~Would it be preferable to only set preConfigure on darwin, or run it everywhere for consistency?~

Tested using, eg, `nix build -f . libtheora` on master, and `nix build -f . --argstr system aarch64-darwin libtheora` on https://github.com/thefloweringash/nixpkgs/tree/apple-silicon

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/development/libraries/libtheora/default.nix:
```diff
@@ -1,4 +1,4 @@
-{lib, stdenv, fetchurl, libogg, libvorbis, pkg-config}:
+{lib, stdenv, fetchurl, libogg, libvorbis, pkg-config, automake, autoconf, libtool}:
```
```suggestion
{ lib, stdenv, fetchurl, libogg, libvorbis, pkg-config, autoreconfHook }:
```

SuperSandro2000 commented on pkgs/development/libraries/libtheora/default.nix:
```diff
@@ -11,7 +11,7 @@ stdenv.mkDerivation rec {
   outputDoc = "devdoc";
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [ pkg-config automake autoconf libtool ];
```
```suggestion
  nativeBuildInputs = [ pkg-config autoreconfHook ];
```

SuperSandro2000 commented on pkgs/development/libraries/libtheora/default.nix:
```diff
@@ -20,6 +20,10 @@ stdenv.mkDerivation rec {
     substituteInPlace configure --replace "-fforce-addr" ""
   '';
 
+  # libtheora's configure script is generated with an old version of libtool.
+  # Regenerate with a modern version:
+  preConfigure = "sh autogen.sh";
+
```
```suggestion
```

autoreconfHook should take care of this.

jdelStrother replied:
Autoconf is starting to make me feel very dumb, I don't think I can fix it. Made your suggested changes but run into errors like:

```
libtheora> unpacking sources
libtheora> unpacking source archive /nix/store/1s6ssdh3sjciqc0lzny58bf0iymbadzy-libtheora-1.1.1.tar.gz
libtheora> source root is libtheora-1.1.1
libtheora> setting SOURCE_DATE_EPOCH to timestamp 1254420253 of file libtheora-1.1.1/config.h.in
libtheora> patching sources
libtheora> autoreconfPhase
libtheora> autoreconf: export WARNINGS=
libtheora> autoreconf: Entering directory '.'
libtheora> autoreconf: configure.ac: not using Gettext
libtheora> autoreconf: running: aclocal --force
libtheora> configure.ac:316: warning: macro 'AM_PATH_SDL' not found in library
libtheora> autoreconf: configure.ac: tracing
libtheora> autoreconf: running: libtoolize --copy --force
libtheora> libtoolize: putting auxiliary files in '.'.
libtheora> libtoolize: copying file './ltmain.sh'
libtheora> libtoolize: Consider adding 'AC_CONFIG_MACRO_DIRS([m4])' to configure.ac,
libtheora> libtoolize: and rerunning libtoolize and aclocal.
libtheora> libtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
libtheora> autoreconf: configure.ac: not using Intltool
libtheora> autoreconf: configure.ac: not using Gtkdoc
libtheora> autoreconf: running: aclocal --force
libtheora> configure.ac:316: warning: macro 'AM_PATH_SDL' not found in library
libtheora> autoreconf: running: /nix/store/1jx7wq4c3d93v585jzlyra0hn99djk6i-autoconf-2.70/bin/autoconf --force
libtheora> configure.ac:12: warning: 'AM_CONFIG_HEADER': this macro is obsolete.
libtheora> configure.ac:12: You should use the 'AC_CONFIG_HEADERS' macro instead.
libtheora> ./lib/autoconf/general.m4:2433: AC_DIAGNOSE is expanded from...
libtheora> aclocal.m4:10322: AM_CONFIG_HEADER is expanded from...
libtheora> configure.ac:12: the top level
libtheora> configure.ac:60: warning: The macro `AC_LIBTOOL_WIN32_DLL' is obsolete.
libtheora> configure.ac:60: You should run autoupdate.
libtheora> aclocal.m4:8769: AC_LIBTOOL_WIN32_DLL is expanded from...
libtheora> configure.ac:60: the top level
libtheora> configure.ac:60: warning: AC_LIBTOOL_WIN32_DLL: Remove this warning and the call to _LT_SET_OPTION when you
libtheora> configure.ac:60: put the 'win32-dll' option into LT_INIT's first parameter.
libtheora> ./lib/autoconf/general.m4:2433: AC_DIAGNOSE is expanded from...
libtheora> aclocal.m4:8769: AC_LIBTOOL_WIN32_DLL is expanded from...
libtheora> configure.ac:60: the top level
libtheora> configure.ac:61: warning: The macro `AM_PROG_LIBTOOL' is obsolete.
libtheora> configure.ac:61: You should run autoupdate.
libtheora> aclocal.m4:376: AM_PROG_LIBTOOL is expanded from...
libtheora> configure.ac:61: the top level
libtheora> configure.ac:208: warning: The macro `AC_PROG_LD' is obsolete.
libtheora> configure.ac:208: You should run autoupdate.
libtheora> aclocal.m4:3598: AC_PROG_LD is expanded from...
libtheora> configure.ac:208: the top level
libtheora> configure.ac:276: warning: The macro `AC_HELP_STRING' is obsolete.
libtheora> configure.ac:276: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
libtheora> aclocal.m4:167: XIPH_PATH_OGG is expanded from...
libtheora> configure.ac:276: the top level
libtheora> configure.ac:276: warning: The macro `AC_TRY_RUN' is obsolete.
libtheora> configure.ac:276: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:2996: AC_TRY_RUN is expanded from...
libtheora> aclocal.m4:167: XIPH_PATH_OGG is expanded from...
libtheora> configure.ac:276: the top level
libtheora> configure.ac:276: warning: The macro `AC_TRY_LINK' is obsolete.
libtheora> configure.ac:276: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:2919: AC_TRY_LINK is expanded from...
libtheora> aclocal.m4:167: XIPH_PATH_OGG is expanded from...
libtheora> configure.ac:276: the top level
libtheora> configure.ac:310: warning: The macro `AC_HELP_STRING' is obsolete.
libtheora> configure.ac:310: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:203: AC_HELP_STRING is expanded from...
libtheora> aclocal.m4:31: XIPH_PATH_VORBIS is expanded from...
libtheora> configure.ac:310: the top level
libtheora> configure.ac:310: warning: The macro `AC_TRY_RUN' is obsolete.
libtheora> configure.ac:310: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:2996: AC_TRY_RUN is expanded from...
libtheora> aclocal.m4:31: XIPH_PATH_VORBIS is expanded from...
libtheora> configure.ac:310: the top level
libtheora> configure.ac:310: warning: The macro `AC_TRY_LINK' is obsolete.
libtheora> configure.ac:310: You should run autoupdate.
libtheora> ./lib/autoconf/general.m4:2919: AC_TRY_LINK is expanded from...
libtheora> aclocal.m4:31: XIPH_PATH_VORBIS is expanded from...
libtheora> configure.ac:310: the top level
libtheora> configure.ac:452: warning: AC_OUTPUT should be used without arguments.
libtheora> configure.ac:452: You should run autoupdate.
libtheora> configure.ac:316: error: possibly undefined macro: AM_PATH_SDL
libtheora> If this token and others are legitimate, please use m4_pattern_allow.
libtheora> See the Autoconf documentation.
libtheora> configure.ac:469: error: possibly undefined macro: AS_AC_EXPAND
libtheora> autoreconf: error: /nix/store/1jx7wq4c3d93v585jzlyra0hn99djk6i-autoconf-2.70/bin/autoconf failed with exit status: 1
```

jdelStrother replied:
I've fixed this with SuperSandro2000's help. Will submit a new patch soon...

jdelStrother commented:
> Tested compilation of all pkgs that depend on this change using nix-shell -p nixpkgs-review --run "nixpkgs-review wip"

I ran this, and got output that doesn't sound great:

```
error: while setting up the build environment: getting attributes of path '/usr/lib/libSystem.B.dylib': No such file or directory
27 packages marked as broken and skipped:
dolphinEmuMaster ghostwriter gnome2.libcanberra-gtk2 gnomecast hydrus libcanberra libcanberra-gtk3 libcanberra_gtk2 libcanberra_gtk3 mirage-im mopidy-musicbox-webclient mpc-qt onionshare-gui opustags privateer ptcollab python38Packages.pyside2 python38Packages.pyside2-tools python38Packages.test-tube python38Packages.tumpa python39Packages.pyside2 python39Packages.pyside2-tools python39Packages.test-tube python39Packages.tumpa qsyncthingtray sl1-to-photon sub-batch

228 packages failed to build:
alass anki bazarr beamerpresenter blender botamusique brewtarget catt cb2bib chatterino2 chiaki chromaprint cmus cool-old-term corrscope cozy curseradio doomseeker dr14_tmeter ezstream fast-cli ffmpeg ffmpeg-full ffmpeg-normalize ffmpeg_2 ffmpeg_3 ffms frostwire get_iplayer gif-for-cli goldendict gopro gst_all_1.gst-devtools gst_all_1.gst-editing-services gst_all_1.gst-libav gst_all_1.gst-plugins-bad gst_all_1.gst-plugins-base gst_all_1.gst-plugins-good gst_all_1.gst-plugins-ugly gst_all_1.gst-rtsp-server handbrake httraqt hydron icecast imgbrd-grabber jellyfin jellyfin_10_5 kristall libgroove libquotient libsForQt5.full libsForQt5.libopenshot libsForQt5.qmltermwidget libsForQt5.qt5.qtlocation libsForQt5.qt5.qtmultimedia libsForQt5.qt5.qtwebengine libsForQt5.qt5.qtwebkit libsForQt5.qt5.qtwebview libsForQt514.full libsForQt514.libopenshot libsForQt514.qmltermwidget libsForQt514.qt5.qtlocation libsForQt514.qt5.qtmultimedia libsForQt514.qt5.qtwebengine libsForQt514.qt5.qtwebkit libsForQt514.qt5.qtwebview libsForQt515.full libsForQt515.libopenshot libsForQt515.qmltermwidget libsForQt515.qt5.qtlocation libsForQt515.qt5.qtmultimedia libsForQt515.qt5.qtwebengine libsForQt515.qt5.qtwebkit libsForQt515.qt5.qtwebview libshout lidarr livepeer lsd2dsl manim mellowplayer merkaartor mindforger mkchromecast mnemosyne mopidy mopidy-iris mopidy-local mopidy-moped mopidy-mopify mopidy-mpd mopidy-mpris mopidy-scrobbler mopidy-somafm mopidy-soundcloud mopidy-spotify mopidy-spotify-tunigo mopidy-subidy mopidy-tunein mopidy-youtube motion mpd mpd-small mplayer mps-youtube mpv mpv-unwrapped musly nethack-qt nmapsi4 notcurses octavePackages.video olive-editor openorienteering-mapper openrw openscad openshot-qt opentoonz ovito pangolin pcsxr phantomjs2 pianobar pidgin pithos plex-media-player plex-mpv-shim purple-matrix python-qt python38Packages.aplpy python38Packages.asdf python38Packages.av python38Packages.batchgenerators python38Packages.dask-image python38Packages.ffmpeg-python python38Packages.glymur python38Packages.gst-python python38Packages.ha-ffmpeg python38Packages.handout python38Packages.image-match python38Packages.imagecorruptions python38Packages.imageio python38Packages.infoqscraper python38Packages.moviepy python38Packages.mpv python38Packages.pafy python38Packages.pims python38Packages.pyacoustid python38Packages.pydub python38Packages.pyqt5_with_qtmultimedia python38Packages.pyqt5_with_qtwebkit python38Packages.pyqtwebengine python38Packages.pyregion python38Packages.pytest-astropy python38Packages.pytest-astropy-header python38Packages.radio_beam python38Packages.reproject python38Packages.runway-python python38Packages.scikitimage python38Packages.slicedimage python38Packages.spectral-cube python38Packages.stytra python38Packages.sunpy python38Packages.tubeup youtubeDL python39Packages.aplpy python39Packages.asdf python39Packages.av python39Packages.batchgenerators python39Packages.dask-image python39Packages.ffmpeg-python python39Packages.glymur python39Packages.gst-python python39Packages.ha-ffmpeg python39Packages.handout python39Packages.image-match python39Packages.imagecorruptions python39Packages.imageio python39Packages.infoqscraper python39Packages.moviepy python39Packages.mpv python39Packages.pafy python39Packages.pims python39Packages.pyacoustid python39Packages.pydub python39Packages.pyqt5_with_qtmultimedia python39Packages.pyqt5_with_qtwebkit python39Packages.pyqtwebengine python39Packages.pyregion python39Packages.pytest-astropy python39Packages.pytest-astropy-header python39Packages.radio_beam python39Packages.reproject python39Packages.runway-python python39Packages.scikitimage python39Packages.slicedimage python39Packages.spectral-cube python39Packages.sunpy python39Packages.tubeup python39Packages.youtube-dl qgo qtbitcointrader qtox quaternion qutebrowser r128gain ripgrep-all sayonara scrcpy sigal sleepyhead somafm-cli stellarium streamlink sumo svtplay-dl tipp10 togglesg-download unpaper untrunc vcs vcsi webmacs wireshark yacreader yle-dl ytcc ytfzf zaz

1 package built:
libtheora

error: build log of '/nix/store/fmqxvs0b5l5dy9ql8c1ngwj1vdi7n0zn-alass-2.0.0.drv' is not available
error: build log of '/nix/store/0hj352i1scg70cxh5gfdykhzih4b2yf5-anki-2.1.15.drv' is not available
error: build log of '/nix/store/6qmz0wbhaslmmkshadspw5qfmbcq962y-bazarr-0.9.2.drv' is not available
error: build log of '/nix/store/klk9rglai3ls7wrg0ck0lr5cxsziqvyf-beamerpresenter-0.1.3.drv' is not available
error: build log of '/nix/store/2068swd2wbw0r7834hqjqpvfqbnrsbi7-blender-2.91.0.drv' is not available
error: build log of '/nix/store/dxpyb6gfp90yrknjiczljnncdyidp8n8-botamusique-unstable-2021-03-13.drv' is not available
error: build log of '/nix/store/h2g6vb345h6376qndshagqvzmx42vcz3-brewtarget-2.3.1.drv' is not available
...
```

but don't know what the expected output is.

If I run, eg, `nix-shell -I nixpkgs=. -p ffmpeg`, ffmpeg builds & runs fine 😕

SuperSandro2000 commented:
> I thought it might be better to submit this separately?

This is a good idea even if it causes more rebuilds because we can easily review the changes and notice small breakages early rather than everything breaking at once if at all.

> I ran this, and got output that doesn't sound great

Please fetch the latest nixpkgs-review with an overlay from https://github.com/mic92/nixpkgs-review. The master branch no longer turns on sandbox on darwin which most likely caused your issue.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 118132` at d55c80cd run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>196 packages marked as broken and skipped:</summary>
  <ul>
    <li>aqemu</li>
    <li>avxsynth</li>
    <li>betaflight-configurator</li>
    <li>bless</li>
    <li>cl</li>
    <li>couchdb</li>
    <li>couchdb2</li>
    <li>cuter</li>
    <li>devede</li>
    <li>erlangR18</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/0b88e3181c098f55e08ba373616f7145">airtame</a></li>
    <li><a href="https://gist.github.com/7ca842f09e18275aa5ced562b222bbf9">jrePlugin</a> (jre8Plugin)</li>
    <li><a href="https://gist.github.com/0a9ae9b4c58b08b02e0d36815d8779b6">kodelife</a></li>
    <li><a href="https://gist.github.com/61cb928b3a89337ef74beae9e8fc8df7">oraclejdk</a> (oraclejdk8)</li>
    <li><a href="https://gist.github.com/81510dfcc55f89e1e92c06d2ae5ad7ab">oraclejre</a> (oraclejre8)</li>
    <li><a href="https://gist.github.com/ff99f25f3e36cd3de2482a58282e5759">svtplay-dl</a></li>
  </ul>
</details>
<details>
  <summary>1150 packages skipped due to time constraints:</summary>
  <ul>
    <li>MIDIVisualizer</li>
    <li>_90secondportraits</li>
    <li>acoustidFingerprinter</li>
    <li>adapta-gtk-theme</li>
    <li>adoptopenjdk-icedtea-web</li>
    <li>ahoviewer</li>
    <li>alarm-clock-applet</li>
    <li>alass</li>
    <li>alephone</li>
    <li>allegro5</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>45 packages built successfully:</summary>
  <ul>
    <li>brasero</li>
    <li>brasero-original</li>
    <li>chromaprint</li>
    <li>corrscope</li>
    <li>dr14_tmeter</li>
    <li>farstream</li>
    <li>ffmpeg (ffmpeg_4)</li>
    <li>ffmpeg-normalize</li>
    <li>ffmpeg_3 (ffmpeg_3_4)</li>
    <li>get_iplayer</li>
    <li>libcanberra-gtk2 (gnome2.libcanberra-gtk2)</li>
    <li>gnome3.gnome-bluetooth</li>
    <li>gnome3.gnome-settings-daemon (gnome3.gnome_settings_daemon)</li>
    <li>gsound (gnome3.gsound)</li>
    <li>gst_all_1.gst-libav</li>
    <li>gst_all_1.gst-plugins-bad</li>
    <li>gst_all_1.gst-plugins-base</li>
    <li>gst_all_1.gst-plugins-good</li>
    <li>gst_all_1.gst-plugins-ugly</li>
    <li>gupnp-dlna</li>
    <li>harvid</li>
    <li>keyfinder-cli</li>
    <li>libcanberra (libcanberra_kde)</li>
    <li>libcanberra-gtk3</li>
    <li>libnice</li>
    <li>qt5.qtmultimedia (libsForQt5.qtmultimedia ,qt515.qtmultimedia)</li>
    <li>libshout</li>
    <li>libtheora</li>
    <li>nice-dcv-client</li>
    <li>pipewire</li>
    <li>python38Packages.gst-python</li>
    <li>python38Packages.ha-ffmpeg</li>
    <li>python38Packages.imageio</li>
    <li>python38Packages.infoqscraper</li>
    <li>python38Packages.moviepy</li>
    <li>python38Packages.pyacoustid</li>
    <li>youtube-dl (python38Packages.youtube-dl)</li>
    <li>python39Packages.ha-ffmpeg</li>
    <li>python39Packages.imageio</li>
    <li>python39Packages.infoqscraper</li>
    <li>roon-server</li>
    <li>togglesg-download</li>
    <li>unpaper</li>
    <li>vcsi</li>
    <li>xjadeo</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

jdelStrother commented:
> Please fetch the latest nixpkgs-review with an overlay from https://github.com/mic92/nixpkgs-review. The master branch no longer turns on sandbox on darwin which most likely caused your issue.

Thanks, got it working with this in ~/.config/nixpkgs/overlays.nix:
```nix
[
  (self: super: {
    nixpkgs-review = super.nixpkgs-review.overrideAttrs (old: {
      version = "2.6.0.rc1.1";
      buildInputs = [(super.python3.withPackages (ps: [ps.pytest]))];
      src = super.fetchFromGitHub {
        owner = "Mic92";
        repo = "nixpkgs-review";
        rev = "9395205ee46c0db722817ac2bc1b4b293779dd2f";
        sha256 = "17aab9wg8a4g4sd0w6lidzrk7r048i9x11328ia2f3ixyx5aipnk";
      };
    });
  })
]

```

jdelStrother committed:
libtheora: regenerate config with autoreconfHook

libtheora's configure script comes from a very old version of libtool.
This fixes it up to work with autoreconf, which should make it
forward-compatible with arm on macOS (where MACOSX_DEPLOYMENT_TARGET=11.0)

jdelStrother commented:
OK, I've pushed an updated version that uses autoreconfHook

SuperSandro2000 commented on pkgs/development/libraries/libtheora/default.nix:
```diff
@@ -7,19 +7,19 @@ stdenv.mkDerivation rec {
     sha256 = "0swiaj8987n995rc7hw0asvpwhhzpjiws8kr3s6r44bqqib2k5a0";
   };
+  patches = [
+    (fetchpatch {
```
```suggestion
    # fix error in autoconf scripts
    (fetchpatch {
```

SuperSandro2000 commented:
Other than that LGTM. I am building it later for darwin and linux.

jdelStrother commented:
> Other than that LGTM. I am building it later for darwin and linux.

thanks!  I've pushed an update with your suggestion

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118132` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>libtheora</li>
</ul>
</details>

SuperSandro2000 commented on pkgs/development/libraries/libtheora/default.nix:
```diff
@@ -7,19 +7,20 @@ stdenv.mkDerivation rec {
     url = "http://downloads.xiph.org/releases/theora/${name}.tar.gz";
     sha256 = "0swiaj8987n995rc7hw0asvpwhhzpjiws8kr3s6r44bqqib2k5a0";
   };
+  patches = [
```
```suggestion
  };

  patches = [
```

SuperSandro2000 committed:
Update pkgs/development/libraries/libtheora/default.nix

SuperSandro2000 commented:
/rebase staging
